Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp

```diff
@@ -354,20 +369,249 @@ bool ISO8601Parser::parse_time_fraction()
+{
+    // TimeZoneUTCOffsetMinute :
+    //     MinuteSecond
+    StateTransaction transaction { *this };
+    if (!parse_minute_second())
+        return false;
+    m_state.parse_result.time_zone_utc_offset_minute = transaction.parsed_string_view();
+    transaction.commit();
+    return true;
+}
+
+// https://tc39.es/proposal-temporal/#prod-TimeZoneUTCOffsetSecond
+bool ISO8601Parser::parse_time_zone_utc_offset_second()
+{
+    // TimeZoneUTCOffsetSecond :
+    //     MinuteSecond
+    StateTransaction transaction { *this };
+    if (!parse_minute_second())
+        return false;
+    m_state.parse_result.time_zone_utc_offset_second = transaction.parsed_string_view();
+    transaction.commit();
+    return true;
+}
+
+// https://tc39.es/proposal-temporal/#prod-TimeZoneUTCOffsetFractionalPart
+bool ISO8601Parser::parse_time_zone_utc_offset_fractional_part()
+{
+    // TimeZoneUTCOffsetFractionalPart :
+    //     FractionalPart
+    StateTransaction transaction { *this };
+    if (!parse_fractional_part())
+        return false;
+    m_state.parse_result.time_zone_utc_offset_fractional_part = transaction.parsed_string_view();
+    transaction.commit();
+    return true;
+}
+
+// https://tc39.es/proposal-temporal/#prod-TimeZoneUTCOffsetFraction
+bool ISO8601Parser::parse_time_zone_utc_offset_fraction()
+{
+    // TimeZoneUTCOffsetFraction :
+    //     DecimalSeparator TimeZoneUTCOffsetFractionalPart
+    StateTransaction transaction { *this };
+    if (!parse_decimal_separator())
+        return false;
+    if (!parse_time_zone_utc_offset_fractional_part())
+        return false;
+    transaction.commit();
+    return true;
+}
+
+// https://tc39.es/proposal-temporal/#prod-TimeZoneNumericUTCOffset
+bool ISO8601Parser::parse_time_zone_numeric_utc_offset()
+{
+    // TimeZoneNumericUTCOffset :
+    //     TimeZoneUTCOffsetSign TimeZoneUTCOffsetHour
+    //     TimeZoneUTCOffsetSign TimeZoneUTCOffsetHour : TimeZoneUTCOffsetMinute
+    //     TimeZoneUTCOffsetSign TimeZoneUTCOffsetHour TimeZoneUTCOffsetMinute
+    //     TimeZoneUTCOffsetSign TimeZoneUTCOffsetHour : TimeZoneUTCOffsetMinute : TimeZoneUTCOffsetSecond TimeZoneUTCOffsetFraction[opt]
+    //     TimeZoneUTCOffsetSign TimeZoneUTCOffsetHour TimeZoneUTCOffsetMinute TimeZoneUTCOffsetSecond TimeZoneUTCOffsetFraction[opt]
+    StateTransaction transaction { *this };
+    if (!parse_time_zone_utc_offset_sign())
+        return false;
+    if (!parse_time_zone_utc_offset_hour())
+        return false;
+    if (m_state.lexer.consume_specific(':')) {
+        if (!parse_time_zone_utc_offset_minute())
+            return false;
+        if (m_state.lexer.consume_specific(':')) {
+            if (!parse_time_zone_utc_offset_second())
+                return false;
+            (void)parse_time_zone_utc_offset_fraction();
+        }
+    } else if (parse_time_zone_utc_offset_minute()) {
+        if (!parse_time_zone_utc_offset_second())
+            return false;
```

Comment:
This doesn't handle the `TimeZoneUTCOffsetSign TimeZoneUTCOffsetHour TimeZoneUTCOffsetMinute` case

Reply:
Fixed!